index.js

```diff
@@ -42,14 +44,30 @@ module.exports = function requireAll(options) {
         filter: filter,
         excludeDirs: excludeDirs,
         map: map,
-        resolve: resolve
+        resolve: resolve,
+        merge: merge
       });
 
     } else {
       var name = filterFile(file);
       if (!name) return;
 
-      modules[map(name, filepath)] = resolve(require(filepath));
+      var module = resolve(require(filepath));
+
+      if (merge) {
+        // console.log('module and is object', module, isObject(module))
+      if (isObject(module)) {
+        for (let key in module) {
+          modules[key] = module[key];
```

Comment:
seem like that would be a bad idea.  Then you'd be messing with the keys of the actual module instead of the key/prop that was to be inserted for the modules (i.e. the filename or some other value from the map).  It does use map for those modules without any props/keys.